### PR TITLE
Update Firestore type names

### PR DESCRIPTION
**Pull request by Feiyang1**
go/js-v9-types-naming

### CODE REVIEW

File: common/api-review/firestore-lite.api.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ```ts
 
+import { EmulatorMockTokenOptions } from '@firebase/util';
```
Comment (schmidt-sebastian) [inline common/api-review/firestore-lite.api.md:7]: Can you rename `Settings` to FirestoreSettings as well (for Lite)?

**Review by schmidt-sebastian — APPROVED** [review]

File: common/api-review/firestore-lite.api.md [hunk 2]
```diff
@@ -116,9 +118,11 @@ export abstract class FieldValue {
 }
 
 // @public
-export class FirebaseFirestore {
+export class Firestore {
     get app(): FirebaseApp;
     toJSON(): object;
+    // (undocumented)
```
[the remark this answers is not in the document]
Comment (Feiyang1) [reply common/api-review/firestore-lite.api.md]: Done.

File: common/api-review/firestore-lite.api.md (hunk 1 above)
Comment (Feiyang1) [reply common/api-review/firestore-lite.api.md:7]: Done

File: packages/firestore/src/lite/database.ts [hunk 3]
```diff
@@ -59,14 +59,17 @@ declare module '@firebase/component' {
  *
  * Do not call this constructor directly. Instead, use {@link getFirestore}.
  */
-export class FirebaseFirestore implements FirestoreService {
+export class Firestore implements FirestoreService {
+  /**
+   * whether it's a firestore or firestore-lite instance
```
Comment (Feiyang1) [inline packages/firestore/src/lite/database.ts:64]: @egilmorez Does it look good to you?

Comment (egilmorez) [reply packages/firestore/src/lite/database.ts:64]: If it's public, we should cap "Whether" and either:

-- backtick the types of instances as literals
-- just use the capitalized product names.

File: packages/firestore/src/exp/database.ts [hunk 4]
```diff
@@ -79,7 +79,10 @@ export const CACHE_SIZE_UNLIMITED = LRU_COLLECTION_DISABLED;
  *
  * Do not call this constructor directly. Instead, use {@link getFirestore}.
  */
-export class FirebaseFirestore extends LiteFirestore {
+export class Firestore extends LiteFirestore {
+  /**
+   * whether it's a firestore or firestore-lite instance
```
Comment (egilmorez) [inline packages/firestore/src/exp/database.ts:84]: Same comment here as below regarding caps/backticks.

But does this "database.ts" imply RTDB, and is this comment in the right place if so?

**Review by egilmorez — COMMENTED** [review]
Thanks for putting me on the review Feiyang!

One meta question: is there a logical way to separate files with publicly-viewable comments from internal-only files?  If so, that might be a good way to narrow down the size/scope of these kinds of PRs.

File: packages/firestore/src/lite/database.ts (hunk 3 above)
Comment (Feiyang1) [reply packages/firestore/src/lite/database.ts]: They share the same type name, so I changed to use the capitalized product names.

**Comment by Feiyang1** [comment]
> Thanks for putting me on the review Feiyang!
> 
> One meta question: is there a logical way to separate files with publicly-viewable comments from internal-only files? If so, that might be a good way to narrow down the size/scope of these kinds of PRs.

In simpler packages like app, analytics, remote-config, all public APIs are defined in `api.ts` and `public-types.ts`, but it's difficult to enforce the same convention for auth, firestore, database that are more complex. 

The best way to identify public docs is looking for the `@public` annotation. I wonder if we can make a GHA that highlights these locations in PRs. I have seen similar things for highlighting lint errors, so it should be doable. I will investigate.

File: packages/firestore/index.node.ts [hunk 5]
```diff
@@ -20,7 +20,7 @@ import { FirebaseNamespace } from '@firebase/app-types';
 import { name, version } from './package.json';
 import { Firestore, IndexedDbPersistenceProvider } from './src/api/database';
 import { configureForFirebase } from './src/config';
-import { FirebaseFirestore as ExpFirebaseFirestore } from './src/exp/database';
+import { Firestore as ExpFirebaseFirestore } from './src/exp/database';
```
Comment (hsubox76) [inline packages/firestore/index.node.ts:23]: I know it's just internal so it doesn't matter that much but for consistency in readability, should remove the `Firebase` from the `ExpFirebaseFirestore` too, right? Saw at least one other occurrence in another file.

**Review by hsubox76 — APPROVED** [review]

File: packages/firestore/index.node.ts (hunk 5 above)
Comment (Feiyang1) [reply packages/firestore/index.node.ts]: updated

File: packages/firestore/src/exp/database.ts [hunk 6]
```diff
@@ -79,7 +79,10 @@ export const CACHE_SIZE_UNLIMITED = LRU_COLLECTION_DISABLED;
  *
  * Do not call this constructor directly. Instead, use {@link getFirestore}.
  */
-export class FirebaseFirestore extends LiteFirestore {
+export class Firestore extends LiteFirestore {
+  /**
+   * Whether it's a Firestore or Firestore Lite instance
```
Comment (egilmorez) [inline packages/firestore/src/exp/database.ts:84]: Missing a period here.

File: packages/firestore/src/lite/database.ts [hunk 7]
```diff
@@ -59,14 +59,17 @@ declare module '@firebase/component' {
  *
  * Do not call this constructor directly. Instead, use {@link getFirestore}.
  */
-export class FirebaseFirestore implements FirestoreService {
+export class Firestore implements FirestoreService {
+  /**
+   * Whether it's a Firestore or Firestore Lite instance
```
Comment (egilmorez) [inline packages/firestore/src/lite/database.ts:64]: Missing a period.

**Review by egilmorez — APPROVED** [review]
LG with nits, thanks!

**Pull request merged**